### ISSUE TITLE
Lab9 Review - Claudio Savelli (S317680)

### ISSUE THREAD

**Issue by ClaudioSavelli** [filippobertolotti0/my_computational_intelligence_317811#3]
Starting with the structure and clarity of the code, the code is very minimal, not very verbose, and easy to read, also because you have commented on each function. Excellent work! 

Furthermore, the README is very explicative and clear and gives an immediate idea of the work done and the procedures followed. It would have been interesting to try to carry out more experiments, perhaps even reporting them in the README file without keeping them only in the notebook. 

As far as the results obtained are concerned, the number of generations to reach the solution in Problem Instance 1 should certainly be emphasised, this is probably due to the lack of dynamism within the code, which always behaves in the same way at any stage of the search, not trying to increase exploration or exploitation when necessary. In addition, the method used to conclude the search could lead to the calling of numerous fitness functions unnecessarily even though it is the purpose of the challenge.

Another interesting point that could be analysed is concerning parent selection, where it might have been interesting to try to choose the parents with higher probability scores (tournament or roulette method). One has a kind of proxy for this behaviour by removing half of the worst parents before the crossover phase. Moreover, the mutation is probably also quite intensive, especially if applied in the same way throughout the whole research phase, and not only in the initial one. On the other hand, I found the use of Gaussian mutation very interesting. The applied crossover is also very interesting. 

In conclusion, the work done is interesting even though it could very easily run into a local minimum. On the other hand, I congratulate you on the work you have done! See you next time!
